src/js/components/Nav/Nav.tsx

```diff
@@ -0,0 +1,139 @@
+
+const navPropTypes = {
+  // React component children
+  children: PropTypes.node,
+  // Custom classnames to add
+  className: PropTypes.string,
+  // TeamSnap UI modifiers to add to class name
+  mods: PropTypes.string,
+  // Custom React Styles to add to the nav
+  style: PropTypes.object,
+  // Any extraneous props
+  otherProps: PropTypes.object,
+  // a list of items to be displayed in the header component
+  headerItems: PropTypes.arrayOf(
+    PropTypes.shape({
+      title: PropTypes.string.isRequired,
+      image: PropTypes.string,
+    })
+  ),
+};
+
+const itemPropTypes = {
+  // String that matches a teamsnap-ui icon name
+  icon: PropTypes.string,
+  // Teamsnap UI modifiers passed to the icon component
+  iconModifiers: PropTypes.string,
+  // bool wether this item should be marked as active or not, multiple can be marked.
+  isActive: PropTypes.bool,
+  // function to handle an item being clicked.
+  onClick: PropTypes.func,
+  // *external* link, creates an anchor tag under the hood. Use onClick and router state for SPA transitions
+  link: PropTypes.string,
+  // any props to pass to the generated anchor tag (target, for example)
+  linkProps: PropTypes.any
+};
+
+type NavType = React.FunctionComponent<
+  PropTypes.InferProps<typeof navPropTypes>
+>;
+type ItemType = React.FunctionComponent<
+  PropTypes.InferProps<typeof itemPropTypes>
+>;
+
+const Item: ItemType = ({ children, icon, iconModifiers, isActive, onClick, link, linkProps }) => {
+  const maybeIcon = icon ? <Icon name={icon} mods={`u-spaceRightXs ${iconModifiers}`} /> : null;
+  const Wrapper = link ? ({children}) => <a href={link} {...linkProps}>{children}</a> : ({children}) => <>{children}</>
+  return (
+    <Wrapper>
+      <li className={`${isActive ? `is-active ` : ``}Nav-item`} onClick={onClick || (() => {})}>
+        {maybeIcon} <span className="Nav-item-title">{children}</span>
+      </li>
+    </Wrapper>
+  );
+};
+
+const Nav: NavType & { Item: ItemType } = ({
+  className,
+  mods,
+  children,
+  style,
+  otherProps,
+  headerItems,
+}) => {
+  const [isCollapsed, setCollapsed] = React.useState(false);
+  const cname = getClassName(
+    "Nav",
+    isCollapsed && "is-collapsed",
+    className,
+    mods
+  );
+
+  const navHeaderIconClass = getClassName(
+    "Nav-header-icon",
+    !isCollapsed && "u-size1of8"
+  );
+
+  return (
+    <nav className={cname} style={style} {...otherProps}>
+      {headerItems && headerItems.length ? (
+        <div className="Nav-header u-textSemiBold">
+          <div
+            className={navHeaderIconClass}
+          >
+            <Avatar
+              src={headerItems[0].image}
+              size="fill"
+            />
+          </div>
+          <div className="u-sizeFill Nav-item-title u-spaceLeftSm">
+            <span className="Nav-item-title u-textEllipsis">
+              {headerItems[0].title}
+            </span>
+          </div>
+          <div className="u-size1of8 Nav-item-title">
```

Comment:
So, if we are always showing down arrow on header, how are we controlling the sub menu when you click on down?

Reply:
We aren't yet. Sorry, this was posted in the project-specific slack channel, but we're going to circle back to this once we know exactly what we want it to do.